Backpack/FloatingNotification/Classes/BPKFloatingNotification.swift

```diff
@@ -0,0 +1,186 @@
+ * limitations under the License.
+ */
+
+import Foundation
+import UIKit
+
+public final class BPKFloatingNotification: UIView {
+    
+    private struct Constants {
+        static let buttonHeight: CGFloat = 20.0
+        static let maxWidth: CGFloat = 400.0
+        static let startBottomConstraint: CGFloat = 30.0
+    }
+    
+    private var onButtonTap: (() -> Void)?
+    private var onDismissal: (() -> Void)?
+    
+    private var bottomConstraint = NSLayoutConstraint()
+    private let animator = FloatingNotificationAnimator()
+    private var notificationsQueue = [FloatingNotificationViewModel]()
+    
+    // MARK: Subviews
+    private let stackView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.distribution = .fillProportionally
+        stackView.alignment = .top
+        stackView.axis = .horizontal
+        stackView.spacing = BPKSpacingMd
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        return stackView
+    }()
+    
+    private let iconView: BPKIconView = {
+        let iconView = BPKIconView(iconName: .none, size: .small)
+        iconView.tintColor = BPKColor.textOnDarkColor
+        iconView.isAccessibilityElement = false
+        iconView.translatesAutoresizingMaskIntoConstraints = false
+        iconView.setContentHuggingPriority(.required, for: .horizontal)
+        iconView.setContentCompressionResistancePriority(.required, for: .horizontal)
+        return iconView
+    }()
+    
+    private let messageLabel: BPKLabel = {
+        let label = BPKLabel()
+        label.fontStyle = .textFootnote
+        label.textColor = BPKColor.textOnDarkColor
+        label.textAlignment = .natural
+        label.numberOfLines = 0
+        label.lineBreakMode = .byWordWrapping
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.setContentHuggingPriority(.defaultLow, for: .horizontal)
+        return label
+    }()
+    
+    private let button: BPKButton = {
+        let button = BPKButton(size: .default, style: .primary)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.setContentHuggingPriority(.required, for: .horizontal)
+        button.setContentCompressionResistancePriority(.required, for: .horizontal)
+        return button
+    }()
+    
+    // MARK: Initialisation
+    public init() {
+        super.init(frame: .zero)
+        button.addTarget(self, action: #selector(buttonTapped), for: .touchUpInside)
+        
+        animator.delegate = self
+        alpha = 0
+        backgroundColor = BPKColor.corePrimaryColor
+        layer.cornerRadius = BPKCornerRadiusMd
+        translatesAutoresizingMaskIntoConstraints = false
+        addSubviews()
+        BPKShadow.shadowLg().apply(to: layer)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    private func addSubviews() {
```

Comment:
Could this be renamed to 'setup()'
then we can also include the 
translatesAutoresizingMaskIntoConstraints = false
and apply shadow from the init function

Reply:
ok sure but will keep `translatesAutoresizingMaskIntoConstraints = false` within the definition of the components as I find it neater